dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -83,25 +81,45 @@ protected CommunityRest createAndReturn(Context context) throws AuthorizeExcepti
         }
 
         Community community;
+        try {
+            // top-level community
+            community = cs.create(null, context);
+            cs.update(context, community);
+            metadataConverter.setMetadata(context, community, communityRest.getMetadata());
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
 
+        return dsoConverter.convert(community);
+    }
+
+    @Override
+    @PreAuthorize("hasPermission(#id, 'COMMUNITY', 'ADD')")
+    protected CommunityRest createAndReturn(Context context, UUID id) throws AuthorizeException {
```

Comment:
I noticed we are missing new ITs to prove the new Collection createAndReturn is checking permissions properly.  They should be similar to the Community ITs you've added, but we should have ITs for both objects.

Reply:
Oops.  That's a big omission.  Tests added.